src/account/accountReducers.js

```diff
@@ -33,7 +33,7 @@ export default (state: AccountState = initialState, action: Action) => {
       ];
     }
     case SET_AUTH_TYPE: {
-      const newState = state.slice();
+      const newState = JSON.parse(JSON.stringify(state));
```

Comment:
i guess, since here https://github.com/zulip/zulip-mobile/blob/master/src/account/accountReducers.js#L37 we were changing property of copied reference(`{authType: 'password'}`) from state. Also verified using deepFreeze.

updated the logic here, to perform faster than `JSON.parse(JSON.stringify(state)` for a large data set.